src/primitives/transaction.h

```diff
@@ -22,8 +22,8 @@ class COutPoint
     uint256 hash;
     uint32_t n;
 
-    COutPoint() { SetNull(); }
-    COutPoint(uint256 hashIn, uint32_t nIn) { hash = hashIn; n = nIn; }
+    COutPoint(): n( (uint32_t) -1 ) { }
+    COutPoint(const uint256& hash, uint32_t n): hash(hash), n(n) { }
```

Comment:
Can you leave the names as `hashIn` and `nIn` so we don't get `hash(hash), n(n)`?